### DISCUSSION TITLE
Heart Rate and Stress onChange not called in simulator

### DISCUSSION THREAD

**Discussion by tip-n** [zepp-health/zeppos-forum#182] [Developer Feedback]
Hello, want to ask about : 
I've tried the code on the zeppos-samples (https://github.com/zepp-health/zeppos-samples)

![image](https://github.com/zepp-health/zeppos-forum/assets/19169445/949ecdfc-37a6-4776-bc6d-f3b8e1b79062)

1. Heart Rate api
- heartRate.getCurrent() is always showing 0
- heartRate.onCurrentChange() never run when i changed heart rate in the simulator
- heartRate.onLastChange() usually run, usually not

my code in page.index.js : 
```
import { HeartRate } from '@zos/sensor'
Page({
    build() {
      const heartRate = new HeartRate()
      const currCallback = () => {
        console.log("current callback, heartRate Current : ", heartRate.getCurrent())
        console.log("current callback, heartRate  Last : ", heartRate.getLast())
      }
      heartRate.onCurrentChange(currCallback)
      const lastCallback = () => {
        console.log("last callback, heartRate Current : ", heartRate.getCurrent())
        console.log("last callback, heartRate Last : ", heartRate.getLast())
      }
      heartRate.onLastChange(lastCallback)
    }
})
```

![image](https://github.com/zepp-health/zeppos-forum/assets/19169445/390bb900-1160-4f9b-b8f6-e7c17f8dcc6b)

2. Stress
It's always showing stress value and time as 0,
also in simulator there's no way to change stress sensor, i've tried changing heart rate but also didn't work
how to test stress sensor using simulator ?

my code in page.index.js : 
```
import { Stress } from '@zos/sensor'
Page({
  build() {
    const stress = new Stress()
    const changeCallback = () => {
      const { value, time } = stress.getCurrent()
      console.log("stress value : ",  value)
      console.log("time value : ", time)
    }
 }
})
```

**Comment by MyoungXUE** [comment]
Thank you for your feedback, we will check as soon as possible.
